src/net/sourceforge/kolmafia/persistence/ConcoctionDatabase.java

```diff
@@ -2065,6 +2065,16 @@ else if (Preferences.getBoolean("requireBoxServants") && !KoLCharacter.inGLover(
     ConcoctionDatabase.CREATION_COST.put(CraftingType.WOOL, 0);
     ConcoctionDatabase.ADVENTURE_USAGE.put(CraftingType.WOOL, 0);
 
+    // Making stuff at The Shadow Forge is only allowed if you are actually there
+    // How can we detect that?
+    if (false) {
```

Comment:
this isn't quite true. You can make stuff at the shadow forge if you haven't spent an adventure since you unlocked it (with `/goto shop.php?whichshop=shadowforge` and probably also with `/buy`).

I was thinking we have a preference `lastForgeUnlockAdventure` and if it matches the current adventure we allow access.

Reply:
OK.

Property ```lastShadowForgeUnlockAdventure``` is now set to KoLCharacter.currentRun when you are about to submit the URL to visit the forge - which will redirect to shop.php

This should enable creation of Shadow Forge items.